Philips.CodeAnalysis.Test/Maintainability/Maintainability/AvoidInvocationAsArgumentTest.cs

```diff
@@ -0,0 +1,318 @@
+class Foo
+{
+  public string Do() { return ""hi"";}
+  public void Moo(string x) { }
+  public string MyTest()
+  {
+     // Comment
+     var renameMe = Do();
+     return Moo(renameMe);
+  }
+}
+";
+
+			VerifyCSharpDiagnostic(errorContent, DiagnosticResultHelper.Create(DiagnosticIds.AvoidInvocationAsArgument));
+			VerifyCSharpFix(errorContent, fixedContent);
+		}
+
+		[TestMethod]
+		public void AvoidInvocationAsArgumentFixIfTest()
+		{
+			string errorContent = @"
+class Foo
+{
+  public string Do() { return ""hi"";}
+  public void Moo(string x) { }
+  public void MyTest()
+  {
+     if (true)
+       Moo(Do());
+  }
+}
+";
+			string fixedContent = @"
+class Foo
+{
+  public string Do() { return ""hi"";}
+  public void Moo(string x) { }
+  public void MyTest()
+  {
+     if (true)
+     {
+       var renameMe = Do();
+       Moo(renameMe);
+     }
+  }
+}
+";
+			VerifyCSharpDiagnostic(errorContent, DiagnosticResultHelper.Create(DiagnosticIds.AvoidInvocationAsArgument));
+			VerifyCSharpFix(errorContent, fixedContent);
+		}
+
+		[TestMethod]
+		public void AvoidInvocationAsArgumentFixNewTest()
+		{
+			string errorContent = @"
+class Meow { public Meow(string x) {} }
+class Foo
+{
+  public string Do() { return ""hi"";}
+  public void MyTest()
+  {
+     new Meow(Do());
+  }
+}
+";
+			string fixedContent = @"
+class Meow { public Meow(string x) {} }
+class Foo
+{
+  public string Do() { return ""hi"";}
+  public void MyTest()
+  {
+     var renameMe = Do();
+     new Meow(renameMe);
+  }
+}
+";
+			VerifyCSharpDiagnostic(errorContent, DiagnosticResultHelper.Create(DiagnosticIds.AvoidInvocationAsArgument));
+			VerifyCSharpFix(errorContent, fixedContent);
+		}
+
+		[TestMethod]
+		public void AvoidInvocationAsArgumentFixSwitchTest()
+		{
+			string errorContent = @"
+class Meow { public Meow(string x) {} }
+class Foo
+{
+  public string Do() { return ""hi"";}
+  public void Moo(string x) { }
+  public void MyTest()
+  {
+     int i;
+     switch (i)
+     {
+       case 0: Moo(Do()); break;
+     }
+  }
+}
+";
+			string fixedContent = @"
+class Meow { public Meow(string x) {} }
+class Foo
+{
+  public string Do() { return ""hi"";}
+  public void Moo(string x) { }
+  public void MyTest()
+  {
+     int i;
+     switch (i)
+     {
+       case 0: var renameMe = Do(); Moo(renameMe); break;
+     }
+  }
+}
+";
+			VerifyCSharpDiagnostic(errorContent, DiagnosticResultHelper.Create(DiagnosticIds.AvoidInvocationAsArgument));
+			VerifyCSharpFix(errorContent, fixedContent);
+		}
+
+		[TestMethod]
+		public void AvoidInvocationAsArgumentFixExpressionTest()
+		{
+			string errorContent = @"
+class Meow { public Meow(string x) {} }
+class Foo
+{
+  public string Do() { return ""hi"";}
+  public void Moo(string x) { }
+  public void MyTest() => Moo(Do());
+}
+";
+
+			VerifyCSharpDiagnostic(errorContent, DiagnosticResultHelper.Create(DiagnosticIds.AvoidInvocationAsArgument));
+			//VerifyCSharpFix(errorContent, fixedContent);
```

Comment:
Could we support Lambda expressions in a future version, if so, please create an Issue for it (or keep this one open)

Reply:
Yes, Issue created.